src/packet/compressed_data.js

```diff
@@ -85,7 +85,7 @@ class CompressedDataPacket {
     await stream.parse(bytes, async reader => {
 
       // One octet that gives the algorithm used to compress the packet.
-      this.algorithm = enums.read(enums.compression, await reader.readByte());
+      this.algorithm = enums.write(enums.compression, await reader.readByte());
```

Comment:
Similarly here - I would just read the byte during parsing, and then if we don't support the algorithm we can complain later, during decompression.